components/script/dom/window.rs

```diff
@@ -1160,11 +1160,17 @@ impl WindowMethods for Window {
 
     // https://html.spec.whatwg.org/multipage/#dom-name
     fn SetName(&self, name: DOMString) {
+        if self.undiscarded_window_proxy().is_none() {
+            return;
+        }
         self.window_proxy().set_name(name);
```

Comment:
Since we've already gotten the `window_proxy` from the previous step, it would be best to reuse it here:

```rust
if let Some(proxy) = self.undiscarded_window_proxy() {
    proxy.set_name(name);
}
```